tests/plugins/os/windows/test_network.py

```diff
@@ -0,0 +1,196 @@
+                "DhcpIPAddress": MockRegVal(name="DhcpIPAddress", value="172.16.0.10"),
+                "DhcpNameServer": MockRegVal(name="DhcpNameServer", value="172.16.0.2"),
+                "DhcpSubnetMask": MockRegVal(name="DhcpSubnetMask", value="255.255.255.0"),
+                "InterfaceMetric": MockRegVal(name="InterfaceMetric", value=5),
+                "Domain": MockRegVal(name="Domain", value="office"),
+            },
+            ["172.16.0.10"],
+            ["172.16.0.2"],
+            ["172.16.0.1"],
+            ["BA:AD:FO:OD:BA:AD"],
+            {
+                "first_connected": datetime.fromisoformat("2012-12-21 00:00:00+00:00"),
+                "type": "IEEE80211",
+                "vlan": "Priority & VLAN Enabled",
+                "name": "Name",
+                "search_domain": "office",
+                "metric": "5",
+                "network": ["255.255.255.0/32"],
+            },
+        ),
+        (
+            {
+                "*PriorityVlanTag": MockRegVal(value="4", name="*PriorityVlanTag"),
+                "NetCfgInstanceId": MockRegVal(value="TESTINGINSTANCEID", name="NetCfgInstanceId"),
+                "*IfType": MockRegVal(value=24, name="*IfType"),
+                "NetworkAddress": MockRegVal(value="FE:ED:FA:CE:FE:ED", name="NetworkAddress"),
+                "NetworkInterfaceInstallTimestamp": MockRegVal(
+                    value=130005216000000000, name="NetworkInterfaceInstallTimestamp"
+                ),
+                "DefaultGateway": MockRegVal(name="DefaultGateway", value="10.0.0.1"),
+                "DhcpIPAddress": MockRegVal(name="DhcpIPAddress", value="10.0.0.10"),
+                "DhcpNameServer": MockRegVal(name="DhcpNameServer", value="10.0.0.2"),
+                "DhcpSubnetMask": MockRegVal(name="DhcpSubnetMask", value="255.255.255.0"),
+                "InterfaceMetric": MockRegVal(name="InterfaceMetric", value=20),
+                "Domain": MockRegVal(name="Domain", value="corp"),
+            },
+            ["10.0.0.10"],
+            ["10.0.0.2"],
+            ["10.0.0.1"],
+            ["FE:ED:FA:CE:FE:ED"],
+            {
+                "first_connected": datetime.fromisoformat("2012-12-21 00:00:00+00:00"),
+                "type": "SOFTWARE_LOOPBACK",
+                "vlan": "OTHER",
+                "name": "Name",
+                "search_domain": "corp",
+                "metric": "20",
+                "network": ["255.255.255.0/32"],
+            },
+        ),
+    ],
+)
+def test_windows_network(
+    mock_values, expected_ips, expected_dns, expected_gateways, expected_macs, expected_interface, target_win: Target
+):
+    with patch("dissect.target.plugins.os.windows.generic.GenericPlugin", return_value=""):
+        with patch("dissect.target.plugins.os.windows._os.WindowsPlugin.hostname", return_value="hostname"):
+            with patch.object(target_win, "registry", MockRegistry(mock_values)):
```

Comment:
```suggestion
           mock_value_dict = dict(map(lambda el: (el[0], MockRegVal(name=el[0], value=el[1])), mock_values.items()))
           with patch.object(target_win, "registry", MockRegistry(mock_value_dict)):
```

I think you can shorten the mock_values by using a map...